pymatgen/io/aims/inputs.py

```diff
@@ -0,0 +1,624 @@
+            if len(split_type) != 2:
+                msg = f"Cube of type {split_type[0]} must have a state associated with it"
+                raise ValueError(msg)
+        else:
+            raise ValueError("Cube type undefined")
+
+        if self.format not in ALLOWED_AIMS_CUBE_FORMATS:
+            raise ValueError("Cube file must have a format of cube, gOpenMol, or xsf")
+
+        if self.spinstate is not None and (self.spinstate not in [1, 2]):
+            raise ValueError("Spin state must be 1 or 2")
+
+        if len(self.origin) != 3:
+            raise ValueError("The cube origin must have 3 components")
+
+        if len(self.points) != 3:
+            raise ValueError("The number of points per edge must have 3 components")
+
+        if len(self.edges) != 3:
+            raise ValueError("Only three cube edges can be passed")
+
+        for edge in self.edges:
+            if len(edge) != 3:
+                raise ValueError("Each cube edge must have 3 components")
+
+        if self.elf_type is not None and self.type != "elf":
+            raise ValueError("elf_type only used when the cube type is elf")
+
+    @property
+    def control_block(self) -> str:
+        """Get the block of text for the control.in file of the Cube"""
+        cb = f"output cube {self.type}\n"
+        cb += f"    cube origin {self.origin[0]: .12e} {self.origin[1]: .12e} {self.origin[2]: .12e}\n"
+        for ii in range(3):
+            cb += f"    cube edge {self.points[ii]} "
+            cb += f"{self.edges[ii][0]: .12e} "
+            cb += f"{self.edges[ii][1]: .12e} "
+            cb += f"{self.edges[ii][2]: .12e}\n"
+        cb += f"    cube format {self.format}\n"
+        if self.spinstate is not None:
+            cb += f"    cube spinstate {self.spinstate}\n"
+        if self.kpoint is not None:
+            cb += f"    cube kpoint {self.kpoint}\n"
+        if self.filename is not None:
+            cb += f"    cube filename {self.filename}\n"
+        if self.elf_type is not None:
+            cb += f"    cube elf_type {self.elf_type}\n"
+
+        return cb
+
+    def as_dict(self) -> dict[str, Any]:
+        """Get a dictionary representation of the geometry.in file."""
+        dct: dict[str, Any] = {}
+        dct["@module"] = type(self).__module__
+        dct["@class"] = type(self).__name__
+        dct["type"] = self.type
+        dct["origin"] = self.origin
+        dct["edges"] = self.edges
+        dct["points"] = self.points
+        dct["format"] = self.format
+        dct["spinstate"] = self.spinstate
+        dct["kpoint"] = self.kpoint
+        dct["filename"] = self.filename
+        dct["elf_type"] = self.elf_type
+        return dct
+
+    @classmethod
+    def from_dict(cls, d: dict[str, Any]) -> AimsCube:
+        """Initialize from dictionary.
+
+        Args:
+            d(dict[str, Any]): The MontyEncoded dictionary
+
+        Returns:
+            The AimsCube for d
+
```

Comment:
del blank line below `Returns:`

Reply:
pyment added those in and I must have missed some